_test/test_combine/test_combine_real.m

```diff
@@ -1,146 +1,149 @@
-            w2d_qq_d2d_minus=read_dnd(fullfile(this.testdir,'w2d_qq_d2d_minus.sqw'));
-            
-            w2d_qq_combined=combine_horace_2d(w2d_qq_d2d_minus,w2d_qq_d2d_plus);
-            
-            mf_qq = multifit_sqw (w2d_qq_d2d);
-            mf_qq = mf_qq.set_fun (@fake_cross_sec, 0.9*[this.stiffness,this.gam,this.amp]);
-            [wfit_qq,fitdata_qq] = mf_qq.fit();
-            
-            mf_combi = multifit_sqw (w2d_qq_combined);
-            mf_combi = mf_combi.set_fun (@fake_cross_sec, 0.9*[this.stiffness,this.gam,this.amp]);
-            [wfit_combi,fitdata_combi] = mf_combi.fit();
-            
-            [ok,mess]=equal_to_tol(fitdata_qq.p,fitdata_combi.p,-2.2e-2,'ignore_str', 1);
-            assertTrue(ok,['combine dnd without a specified tolerance fails: ',mess])
-            
-        end
-        
-        % ------------------------------------------------------------------------------------------------
-        function this = test_combine_dnd_tol(this)
-            % dnd combination specifying a tolerance
-            w1d_d1d=read_dnd(fullfile(this.testdir,'w1d_d1d.sqw'));
-            w1d_d1d_plus=read_dnd(fullfile(this.testdir,'w1d_d1d_plus.sqw'));
-            w1d_d1d_minus=read_dnd(fullfile(this.testdir,'w1d_d1d_minus.sqw'));
-            
-            w1d_combined=combine_horace_1d(w1d_d1d_minus,w1d_d1d_plus,0.025);
-
-            mf = multifit_sqw (w1d_d1d);
-            mf = mf.set_fun (@fake_cross_sec, 0.9*[this.stiffness,this.gam,this.amp]);
-            [wfit,fitdata] = mf.fit();
-            
-            mf_combi = multifit_sqw (w1d_combined);
-            mf_combi = mf_combi.set_fun (@fake_cross_sec, 0.9*[this.stiffness,this.gam,this.amp]);
-            [wfit_combi,fitdata_combi] = mf_combi.fit();
-            
-            [ok,mess]=equal_to_tol(fitdata.p,fitdata_combi.p,-2e-2,'ignore_str', 1);
-            assertTrue(ok,['combine dnd with a specified tolerance fails: ',mess])
-            
-        end        
-    end
-end
+classdef test_combine_real < TestCase
+    % Validate the dnd combination routines
+
+    properties
+        % What we actually want to do is to simulate some cross-section that is
+        % symmetric so that we can compare results easily.
+        stiffness=80;
+        gam=0.1;
+        amp=10;
+        testdir;
+    end
+    
+    methods
+        function this=test_combine_real(name)
+            if ~exist('name','var')
+                name = 'test_combine_real';
+            end
+            this=this@TestCase(name);
+            this.testdir = fileparts(mfilename('fullpath'));
+        end
+        
+        function this=prepare_test_data(this)
+            % Create the data (should not need to do this again)
+            % Use sqw file on RAE's laptop to perform tests. Data saved to a .mat file on SVN server
+            % for validation by others.
+            data_source='C:\Russell\PCMO\ARCS_Oct10\Data\SQW\ei140.sqw';
```

Comment:
Why does this refer explicitly to a file on Russell's computer, this probably shouldn't be in the source.

Reply:
Well. its relatively big file, which was (and is) not in the repository. Its now on SAN so one can get it -- the description there states why it needed. probably I should modify this file with reference to SAN too.